engines/conjur_audit/app/controllers/conjur_audit/messages_controller.rb

```diff
@@ -65,7 +65,16 @@ def messages
         dataset = dataset.limit(limit)
       end
 
-      @messages ||= dataset.all
+      @messages ||= dataset
+    end
+
+    # Convert messages to JSON.
+    # Note: For performance, we use JSON.dump directly.
```

Comment:
Thanks for adding these remarks on why this improves the performance. TIL!

Reply:
FWIW, I found it by profiling. There is another optimization opportunity that should give x20 improvement relating to timestamp handling, but I feel it's good enough for now that fetching 1e5 records takes ~10 seconds.